tests/sles4sap/sap_deployment_automation_framework/deploy_sap_systems.pm

```diff
@@ -75,6 +78,11 @@ sub run {
     load_os_env_variables();
 
     my @installed_components = split(',', get_required_var('SDAF_DEPLOYMENT_SCENARIO'));
+    # This section is only needed by Azure tests using images uploaded
+    if (get_var('PUBLIC_CLOUD_IMAGE_LOCATION')) {
+        my $provider = $self->provider_factory();
+        set_var('SDAF_SOURCE_IMAGE_ID', $self->{provider}->get_image_id());
```

Comment:
This one seems a way to create a sort of "implicit input argument" for `set_image_parameters` : you set something here to let the value to pop up within the function. I personally not like it so much.

Reply:
Thanks for the suggestion. IMO `set_var('SDAF_SOURCE_IMAGE_ID', $self->{provider}->get_image_id());` is simple and easy. Your suggestion is a high standard coding style but a little bit complex.
We can improve it later.